### PR TITLE
Fix docs freeze

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- replace apollo-client with a simple fetch() to get the docs data
- remove onLoad requirement for docs code
- refactor docs and compositions url creator (with tests!)

### CODE REVIEW

File: scopes/compositions/compositions/compositions.tsx [hunk 1]
```diff
@@ -37,7 +38,7 @@ export function Compositions() {
   // collapse sidebar when empty, reopen when not
   useEffect(() => setSidebarOpenness(component.compositions.length > 0), [component.compositions.length]);
 
-  const compositionUrl = `${component.server.url}/#${component.id.fullName}?preview=compositions&`;
+  const compositionUrl = toPreviewUrl(component, 'compositions');
```
Comment (KutnerUri) [inline scopes/compositions/compositions/compositions.tsx:41]: fix - server.url used to be "null"

**Review by odedre — APPROVED** [review]

File: scopes/component/component/ui/component-model/component-model.ts [hunk 2]
```diff
@@ -12,19 +12,22 @@ import { TagProps } from '../../tag/tag';
...
-  status: any; // workspace aspect.
-  deprecation: DeprecationInfo; // deprecation aspect
-  env: Descriptor; // env aspect.
-  labels: string[];
+  issuesCount?: number; // component/issues aspect
+  status?: any; // workspace aspect.
+  deprecation?: DeprecationInfo; // deprecation aspect
+  env?: Descriptor; // env aspect.
+  labels?: string[];
 };
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/component/component/ui/component-model/component-model.ts:31]: this forces our code to be more defensive. it's the opposite of dangerous

File: scopes/preview/ui/component-preview/urls.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import { ComponentModel } from '@teambit/component';
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/preview/ui/component-preview/urls.ts:1]: ok

File: scopes/react/react/docs/use-fetch-docs.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    required: boolean;
+    type: string;
+    default: {
+      value: string;
+    };
+  };
+};
+
+type FetchDocsObj =
+  | {
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/react/react/docs/use-fetch-docs.ts:75]: it's TS syntax.
```tsx
"hello" | "world"
```

is equivalent to 
```tsx
| "hello"
| "world"
```

**Pull request merged**